modules/realm_core/src/worker_thread_base.cpp

```diff
@@ -101,6 +101,19 @@ void WorkerThreadBase::run()
                "Timing [Total]: %lu ms",
                getCurrentTimeMilliseconds() - t);
     }
+
+    // Update statistics for processing
+    std::unique_lock<std::mutex> stat_lock(m_mutex_statistics);
+    long timing = getCurrentTimeMilliseconds() - t;
+    if (m_process_statistics.count == 0) {
+      m_process_statistics.min = (double)timing;
+      m_process_statistics.max = (double)timing;
+    } else {
+      if (m_process_statistics.min > (double)timing) m_process_statistics.min = (double)timing;
+      if (m_process_statistics.max < (double)timing) m_process_statistics.max = (double)timing;
+    }
+    m_process_statistics.count++;
+    m_process_statistics.avg = (m_process_statistics.avg + (double)timing) / 2.0;
```

Comment:
This weighs a single process timing equally to all prior ones averaged. I guess that is not what you intended? Should be more like a floating average: 
```m_process_statistics.avg * (m_process_statistics.count - 1)/m_process_statistics.count + timing * 1/m_process_statistics.count```

Reply:
Yeah, I just went with a quick average to get a feel for timing.  I should probably replace it with a more accurate version.  I'll put that correction in place.

The average equation should simplify to the following to get the same result:
```
m_process_statistics.avg + (timing - m_process_statistics.avg) / m_process_statistics.count
```